src/nomad_material_processing/__init__.py

```diff
@@ -209,6 +505,35 @@ class SubstrateCrystalProperties(CrystalProperties):
         Section describing any miscut of the substrate with respect to the substrate
         orientation.
         """,
+        repeats=True,
+    )
+
+
+class ElectricProperties(ArchiveSection):
+    """
+    The electrical properties of a material.
+    """
+
+    m_def = Section()
+
+    conductivity_type = Quantity(
+        type=MEnum(
+            'P-type',
+            'N-type',
+        ),
+        description='The type of conductivity of the material.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.EnumEditQuantity,
+        ),
+    )
+    carrier_density = Quantity(
+        type=np.dtype(float),
+        unit='1 / cm**3',
+        shape=['*'],
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+        ),
+        description='Concentration of free charge carriers, electrons in the conduction band and holes in the valence band.',
```

Comment:
How do you store this as an array? Is it a tuple of length 2 where the first element is electron concentration and the second is hole? Or is it a function of energy? or k?

Reply:
solved